controllers/llm_controller.go

```diff
@@ -47,10 +57,36 @@ type LLMReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.12.2/pkg/reconcile
 func (r *LLMReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	logger := log.FromContext(ctx)
+	logger.Info("Reconciling LLM resource")
 
-	// TODO(user): your logic here
+	// Fetch the LLM instance
+	instance := &arcadiav1alpha1.LLM{}
+	err := r.Get(ctx, req.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// LLM instance has been deleted.
+			return reconcile.Result{}, nil
+		}
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
 
+	if !instance.DeletionTimestamp.IsZero() {
+		// Instance is being deleted
+		logger.Info("Instance is being deleted")
+		err = r.DeleteLLM(ctx, logger, instance)
```

Comment:
well... when user delete this `LLM` CRD, it has been deleted. So this place should not delete it again.